challenge2/index.html

```diff
@@ -1,74 +1,223 @@
-                            <div class="input-wrapper">
-                                <img src="./assets/icons/lock.svg" alt="lock">
-                                <input type="password" id="passwordConfirm" name="passwordConfirm" placeholder="Confirm your password" required>
-                                <img src="./assets/icons/eye.svg" alt="eye" class="eyeImage">
-                            </div>
-                        </div>
-
-                        <button type="submit">Register</button>
-                    </form>
+              <div class="input-group">
+                <label for="passwordConfirm" id="confirmLabel"
+                  >Confirm Password</label
+                >
+                <div class="input-wrapper" id="confirmWrapper">
+                  <img src="./assets/icons/lock.svg" alt="lock" />
+                  <input
+                    type="password"
+                    id="passwordConfirm"
+                    name="passwordConfirm"
+                    placeholder="Confirm your password"
+                    required
+                  />
+                  <img
+                    src="./assets/icons/eye.svg"
+                    alt="eye"
+                    class="eyeImage"
+                  />
                 </div>
-            </div>
-        </section>
+                <span class="error-message" id="passwordConfirmError"></span>
+              </div>
+
+              <button type="submit">Register</button>
+            </form>
+          </div>
+        </div>
+      </section>
     </main>
 
-</body>
+    <script>
+      document.querySelectorAll(".eyeImage").forEach((eyeIcon) => {
+        eyeIcon.addEventListener("click", function () {
+          const passwordInput = this.previousElementSibling;
+          if (passwordInput.type === "password") {
+            passwordInput.type = "text";
+          } else {
+            passwordInput.type = "password";
+          }
+        });
+      });
+
+      document
+        .getElementById("signupForm")
+        .addEventListener("submit", function (event) {
+          event.preventDefault();
+          validateForm();
+        });
+
+      function validateForm() {
+        const email = document.getElementById("email").value;
+        const username = document.getElementById("username").value;
+        const password = document.getElementById("password").value;
+        const passwordConfirm =
+          document.getElementById("passwordConfirm").value;
+
+        let isValid = true;
+
+        resetErrors();
+
+        // Email validation
+        const emailRegex =
+          /^[a-zA-Z0-9._%+-]+@(gmail\.com|hotmail\.com|lightit\.io)$/;
+        if (!emailRegex.test(email)) {
+          document.getElementById("emailError").textContent =
+            "Invalid email address. Only @gmail.com, @hotmail.com, and @lightit.io are allowed.";
+          turnRed("email");
+          isValid = false;
+        }
```

Comment:
Instead of using comments for these blocks, I think it is cleaner to move them to a different functions. You can use the function names like `validateEmail` to avoid using comments.

This will also help make the `validateForm` function shorter, now it's a bit too long which makes it harder to read.